fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -244,6 +249,28 @@ public void processAlterMaterializedView(AlterMaterializedViewStatement stmt)
         } finally {
             db.writeUnlock();
         }
+        if (materializedView.getRefreshScheme().getType() == MaterializedView.RefreshType.ASYNC) {
+            //drop task
+            TaskManager taskManager = GlobalStateMgr.getCurrentState().getTaskManager();
```

Comment:
if there are some tasks exist in for this MV, why could we directly drop it?

Reply:
yes you can. dropTasks API will kill running TaskRun and remove all pending TaskRun. for PERIODICAL Task it will first stop scheduler.